new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CleanupUtil.java

```diff
@@ -0,0 +1,524 @@
+          logger.info("Service Accounts still exists!!!");
+          List<V1ServiceAccount> items = Kubernetes.listServiceAccounts(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list service accounts");
+      }
+
+      // check if ingress exist
+      try {
+        if (!Kubernetes.listIngressExtensions(namespace).getItems().isEmpty()) {
+          logger.info("Ingress Extensions still exists!!!");
+          List<ExtensionsV1beta1Ingress> items = Kubernetes.listIngressExtensions(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list Ingress Extensions");
+      }
+
+      // check if namespaced roles exist
+      try {
+        if (!Kubernetes.listNamespacedRole(namespace).getItems().isEmpty()) {
+          logger.info("Namespaced roles still exists!!!");
+          List<V1Role> items = Kubernetes.listNamespacedRole(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list namespaced roles");
+      }
+
+      // check if namespaced role bindings exist
+      try {
+        if (!Kubernetes.listNamespacedRoleBinding(namespace).getItems().isEmpty()) {
+          logger.info("Namespaced role bindings still exists!!!");
+          List<V1RoleBinding> items = Kubernetes.listNamespacedRoleBinding(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list namespaced role bindings");
+      }
+
+      // get namespaces
+      try {
+        if (Kubernetes.listNamespaces().contains(namespace)) {
+          logger.info("Namespace still exists!!!");
+          List<String> items = Kubernetes.listNamespaces();
+          for (var item : items) {
+            logger.info(item);
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list namespaces");
+      }
+
+      return doesnotExist;
+    };
+
+  }
+
+  /**
+   * Deletes artifacts in the Kubernetes cluster in the given namespace.
+   *
+   * @param namespace name
+   */
+  public static void deleteNamespacedArtifacts(String namespace) {
```

Comment:
i'd be stunned if this never threw an exception or failed

Reply:
I am catching the exceptions in each delete operation. If I don't catch it inside this method then this method will end prematurely and the delete will be only partial.